lib/nav-menu-builder.ts

```diff
@@ -0,0 +1,23 @@
+export function getSections(baseUrl: string, json: JSON): any[] {
+  let sections: {
+    name: string;
+    items: { title: string; href: string; description: string }[];
+  }[] = [];
+  let listItems: { title: string; href: string; description: string }[] = [];
+
+  Object.entries(json).map(([key, data]: [string, any]) => {
+    listItems = [];
+    Object.entries(data).map(([key, data]: [string, any]) => {
+      listItems.push({
+        title: data.title,
+        href: baseUrl + key,
+        description: data.description,
+      });
+    });
```

Comment:
you don't need to call `.push` from inside of a map...

`Array.map` returns an array.

```js
Object.entries(data).map([key, data] => ({title, href, description}))```